app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/AccountController.scala

```diff
@@ -57,7 +59,15 @@ class AccountController @Inject()(
       _ <- businessEntityJourneyOpt.fold(store.put(BusinessEntityJourney.fromUndertakingOpt(retrievedUndertaking)))(Future.successful)
       _ <- if (retrievedUndertaking.isDefined) store.put(retrievedUndertaking.getOrElse(sys.error("Undertaking is Missing"))) else Future.successful(Unit)
     } yield (retrievedUndertaking, eligibilityJourney, undertakingJourney) match {
-      case (Some(undertaking), _, _) => Ok(accountPage(undertaking, !undertaking.getAllNonLeadEORIs().isEmpty))
+
+      case (Some(undertaking), _, _) =>
+
+        val lastDayToReportDate = undertaking.lastSubsidyUsageUpdt.map(_.plusDays(90))
+        val lastDayToReportString = lastDayToReportDate.map(TimeUtils.govDisplayFormat)
+        val currentDate = timeProvider.today
+        val isTimeToReport = undertaking.lastSubsidyUsageUpdt.map(date => (date.plusDays(75).isBefore(currentDate) && (date.plusDays(91).isAfter(currentDate))))
```

Comment:
You could move this out into a separate helper (a bit like I did with `TaxYearHelpers`) so you can easily unit test this logic in isolation.

Reply:
This is be used once only as of now , will separate it if used at other place. what do you think?

Reply:
Hiding it behind a helper would be good and could hide the option issue pointed out in the template below. Could you also make the 75/91 well named constants of that component (Or configurable if needs be)

Reply:
PR comments addressed